docs/index.md

```diff
@@ -0,0 +1,168 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "Opensearch Provider"
+subcategory: ""
+description: |-
+  The provider is used to interact with the resources supported by Opensearch.
+---
+
+# Opensearch Provider
+
+The provider is used to interact with the resources supported by Opensearch.
+The provider needs to be configured with an endpoint URL before it can be
+used.
+
+AWS Opensearch Service domains and OpenSearch clusters deployed on Kubernetes
+and other infrastructure are supported.
+
+Use the navigation to the left to read about the available resources.
+
+```terraform
+# Configure the Elasticsearch provider
+provider "elasticsearch" {
+  url = "http://127.0.0.1:9200"
+}
+
+# Create an index template
+resource "elasticsearch_index_template" "template_1" {
```

Comment:
should be as [opensearch_index_template](https://github.com/opensearch-project/terraform-provider-opensearch/blob/main/provider/provider.go#L217)